src/main/java/org/openlmis/notification/repository/UserContactDetailsRepository.java

```diff
@@ -39,4 +44,10 @@ default UserContactDetails findOneByEmailAddress(String email) {
     return page.hasContent() ? page.getContent().get(0) : null;
   }
 
+  @Modifying
+  @Transactional
```

Comment:
Why it needs `@Transactional`? Generally, we should keep `@Transactional` limited to service layer only.

Reply:
fixed